uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/controllers/leave-request-popup-controller.js

```diff
@@ -4,34 +4,40 @@ define([
   'common/moment',
   'common/services/api/option-group',
   'common/services/hr-settings',
+  'common/models/contact',
   'leave-absences/shared/models/absence-period-model',
   'leave-absences/shared/models/absence-type-model',
-  'leave-absences/shared/models/entitlement-model',
   'leave-absences/shared/models/calendar-model',
+  'leave-absences/shared/models/entitlement-model',
   'leave-absences/shared/models/leave-request-model',
   'leave-absences/shared/models/public-holiday-model',
   'leave-absences/shared/models/instances/leave-request-instance',
 ], function (components, _, moment) {
   'use strict';
 
   components.controller('LeaveRequestPopupCtrl', [
-    '$log', '$q', '$rootScope', '$uibModalInstance', 'AbsencePeriod', 'AbsenceType',
-    'api.optionGroup', 'contactId', 'Calendar', 'Entitlement', 'HR_settings',
+    '$log', '$q', '$rootScope', '$uibModalInstance', 'Contact', 'AbsencePeriod', 'AbsenceType',
+    'api.optionGroup', 'directiveOptions', 'Calendar', 'Entitlement', 'HR_settings',
     'LeaveRequest', 'LeaveRequestInstance', 'PublicHoliday',
-    function ($log, $q, $rootScope, $modalInstance, AbsencePeriod, AbsenceType,
-      OptionGroup, contactId, Calendar, Entitlement, HR_settings,
+    function ($log, $q, $rootScope, $modalInstance, Contact, AbsencePeriod, AbsenceType,
+      OptionGroup, directiveOptions, Calendar, Entitlement, HR_settings,
       LeaveRequest, LeaveRequestInstance, PublicHoliday
     ) {
       $log.debug('LeaveRequestPopupCtrl');
 
-      var vm = {},
-        serverDateFormat = 'YYYY-MM-DD';
+      var selectedAbsenceType = {},
+        serverDateFormat = 'YYYY-MM-DD',
+        vm = {};
 
       vm.absenceTypes = [];
       vm.calendar = {};
+      vm.contact = {};
```

Comment:
null is for assignment but here we are specifying that it is declared but not yet assigned value